docs/actions-dotnet.md

```diff
@@ -0,0 +1,104 @@
+<!--
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+-->
+
+## Creating and invoking .NET Core actions
+
+The following sections guide you through creating and invoking a single .NET Core action.
+
+In order to compile, test and archive .NET Core projects, you must have the [.NET Core SDK](https://www.microsoft.com/net/download) installed locally and the environment variable `DOTNET_HOME` set to the location where the `dotnet` executable can be found.
+
+A .NET Core action is a .NET Core class library with a method called `Main` that has the exact signature as follows:
+
+```csharp
+public Newtonsoft.Json.Linq.JObject Main(Newtonsoft.Json.Linq.JObject);
+```
+
+For example, create a C# project called `Apache.OpenWhisk.Example.Dotnet`:
+
+```bash
+dotnet new classlib -n Apache.OpenWhisk.Example.Dotnet -lang C#
```

Comment:
the `#` here needs to be escaped or wrapped in quotes otherwise interpreted as a comment in some terminals.